extensions/amp-brightcove/1.0/component.js

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '#preact';
+import {useCallback, useMemo, useRef, useState} from '#preact';
+import {forwardRef} from '#preact/compat';
+import {VideoIframe} from '../../amp-video/1.0/video-iframe';
+import {mutedOrUnmutedEvent} from '../../../src/iframe-video';
+import {VideoEvents} from '../../../src/video-interface';
+import {dispatchCustomEvent} from '#core/dom';
+import {getBrightcoveIframeSrc} from '../brightcove-api';
+
+/** @const {string} */
+const DEFAULT = 'default';
+
+/**
+ * @enum {string}
+ * @private
+ */
+const methods = {
+  'mute': 'muted',
+  'unmute': 'muted',
+  'showControls': 'controls',
+  'hideControls': 'controls',
+};
+
+/**
+ * @enum {string}
+ * @private
+ */
+const events = {
+  'ready': VideoEvents.LOAD,
+  'playing': VideoEvents.PLAYING,
+  'pause': VideoEvents.PAUSE,
+  'ended': VideoEvents.ENDED,
+  'ads-ad-started': VideoEvents.AD_START,
+  'ads-ad-ended': VideoEvents.AD_END,
+  'loadedmetadata': VideoEvents.LOADEDMETADATA,
+};
```

Comment:
The documentation is not clear about `VideoEvents.LOAD` in Classic being the same as dispatching `canplay` on Bento. I'll be adding a section to the guide about these equivalents across versions.

I would suggest removing the entry for `'ready': VideoEvents.LOAD`, and share that enum in `brightcove-api.js` like `amp-vimeo` does. This requires adding a special condition for `ready` to `LOAD` in the Classic version, of course.

Reply:
Thanks for the clarification, done.